File: week1/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    int quarters, dimes, nickels, pennies, coins;
    float dollars, cents, cents_left;

    do
    {
        dollars = get_float("Change owed: ");
    }
    while (dollars < 0);

    cents = round(dollars * 100);
    cents_left = cents;
    //printf("%.0f cents", cents);
    //printf("%.2f dollars ", dollars);

    for (quarters = 0; cents_left >= 25; quarters++)
    {
        cents_left = cents_left - 25;
    }
    //printf("%d quarters", quarters);
    //printf("%f cents left", cents_left);

    for (dimes = 0; cents_left >= 10; dimes++)
    {
        cents_left = cents_left - 10;
    }

    for (nickels = 0; cents_left >= 5; dimes++)
    {
        cents_left = cents_left - 5;
    }

    for (pennies = 0; cents_left >= 1; dimes++)
    {
        cents_left = cents_left - 1;
    }

    coins = quarters + dimes + nickels + pennies; // + nickels + pennies;
    printf("%d\n", coins);
    //printf("%d coins", coins);
    //printf("%f cents left", cents_left);

}

File: week6/cash.py
from cs50 import get_float

# Validates the input
while True:
    dollars = get_float("Change owed: ")
    # print(dollars)
    if dollars > 0.0:
        break

# converts the dollars into cents
cents = round(dollars * 100)
cents_left = cents
coins = 0

# determine how many quarters
while cents_left >= 25:
    cents_left -= 25
    coins += 1

# determine how many dimes
while cents_left >= 10:
    cents_left -= 10
    coins += 1

# determine how many nickels
while cents_left >= 5:
    cents_left -= 5
    coins += 1

# determine how many pennies
while cents_left >= 1:
    cents_left -= 1
    coins += 1

# print coins which is the total number
print(coins)
